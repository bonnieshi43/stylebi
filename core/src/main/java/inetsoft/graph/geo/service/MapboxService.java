/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.graph.geo.service;

import inetsoft.sree.SreeEnv;

import javax.json.*;
import javax.net.ssl.HttpsURLConnection;
import java.awt.*;
import java.io.IOException;
import java.net.URL;
import java.net.URLConnection;
import java.util.ArrayList;
import java.util.List;

public class MapboxService extends WebMapService {
   public MapboxService() {
      user = SreeEnv.getProperty("mapbox.user");
      token = SreeEnv.getProperty("mapbox.token");
      style = SreeEnv.getProperty("mapbox.style");
   }

   public MapboxService(boolean orgScoped) {
      user = SreeEnv.getProperty("mapbox.user", false, orgScoped);
      token = SreeEnv.getProperty("mapbox.token", false, orgScoped);
      style = SreeEnv.getProperty("mapbox.style", false, orgScoped);
   }

   public MapboxService(String user, String token, String style) {
      this.user = user;
      this.token = token;
      this.style = style;
   }

   /**
    * Load a static image using bbox.
    */
   @Override
   public Image loadImage(double lonMin, double lonMax, double latMin, double latMax,
                          int imgw, int imgh)
         throws IOException
   {
      // mapbox static image api:
      // https://api.mapbox.com
      // /styles/v1/{username}/{style_id}/static/{bbox}/{width}x{height}?access_token=...
      String urlstr = String.format("https://api.mapbox.com" +
                                    "/styles/v1/%s/%s/static/[%f,%f,%f,%f]/%dx%d?access_token=%s",
                                 user, style, lonMin, latMin, lonMax, latMax, imgw, imgh, token);
      return WebMapService.loadFromURL(urlstr);
   }

   /**
    * Get a list of all map styles for the user.
    */
   public List<MapboxStyle> getStyles(boolean withDefault) throws IOException {
      List<MapboxStyle> styles = new ArrayList<>();
      getStyles("https://api.mapbox.com/styles/v1/" + user + "?access_token=" + token, styles);

      if(withDefault) {
         styles.add(0, MapboxStyle.DEFAULT);
      }

      return styles;
   }

   private void getStyles(String urlstr, List<MapboxStyle> styles) throws IOException {
      URL url = new URL(urlstr);
      URLConnection conn = url.openConnection();

      try {
         String next = conn.getHeaderField("next");
         JsonReader parser = Json.createReader(conn.getInputStream());
         JsonArray arr = (JsonArray) parser.read();

         arr.stream().map(v -> new MapboxStyle(v)).forEach(v -> styles.add(v));

         if(next != null) {
            getStyles(next, styles);
         }
      }
      catch(IOException ex) {
         if(conn instanceof HttpsURLConnection) {
            if(((HttpsURLConnection) conn).getResponseCode() == 403) {
               throw new RuntimeException("Access token requires permission to STYLES:LIST secret scope");
            }
         }

         throw ex;
      }
   }

   /**
    * Get the tile size. One tile is required to show world at lowest zoom level.
    */
   @Override
   public int getTileSize() {
      return 512;
   }

   @Override
   public int getMaxSize() {
      return 1280;
   }

   @Override
   public boolean isDiscreteZoom() {
      return false;
   }

   private String user;
   private String token;
   private String style;
}
