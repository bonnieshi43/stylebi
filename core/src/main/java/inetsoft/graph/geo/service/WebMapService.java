/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.graph.geo.service;

import inetsoft.graph.coord.GeoCoord;
import inetsoft.sree.SreeEnv;
import inetsoft.uql.viewsheet.graph.MapInfo;
import inetsoft.util.MessageException;

import javax.imageio.ImageIO;
import javax.net.ssl.HttpsURLConnection;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.IOException;
import java.io.Serializable;
import java.net.URL;
import java.net.URLConnection;

/**
 * This is the interface for web map services.
 */
public abstract class WebMapService implements Serializable {
   static BufferedImage loadFromURL(String urlstr) throws IOException {
      URLConnection conn = null;

      try {
         URL url = new URL(urlstr);
         conn = url.openConnection();
         return ImageIO.read(conn.getInputStream());
      }
      catch(IOException ex) {
         if(conn instanceof HttpsURLConnection) {
            int code = ((HttpsURLConnection) conn).getResponseCode();

            if(code == 402) {
               throw new WebMapLimitException();
            }
            else if(code == 403) {
               throw new MessageException("Web map access key is invalid: " + ex.getMessage());
            }
         }

         throw ex;
      }
   }

   /**
    * Get the current web map service.
    * @param mapboxStyle mapbox style or null to use the default.
    */
   public static WebMapService getWebMapService(String mapboxStyle) {
      String service = SreeEnv.getProperty("webmap.service");

      if(MapInfo.MAPBOX.equals(service)) {
         String mapboxUser = SreeEnv.getProperty("mapbox.user");
         String mapboxToken = SreeEnv.getProperty("mapbox.token");

         if(mapboxStyle == null) {
            mapboxStyle = SreeEnv.getProperty("mapbox.style");
         }

         return new MapboxService(mapboxUser, mapboxToken, mapboxStyle);
      }
      else if(MapInfo.GOOGLE.equals(service)) {
         return new GoogleMapsService();
      }

      return null;
   }

    /**
    * Load a static image using bbox.
    */
   public abstract Image loadImage(double lonMin, double lonMax, double latMin, double latMax,
                   int imgw, int imgh)
      throws IOException;

   /**
    * Get the tile size. One tile is required to show world at lowest zoom level.
    */
   public abstract int getTileSize();

   /**
    * Get the maximum size for the image.
    */
   public abstract int getMaxSize();

   /**
    * Check if this web map service only support discrete zoom levels or arbitrary zoom.
    */
   public abstract boolean isDiscreteZoom();

   /**
    * Get the maximum zoom level for discrete zoom.
    */
   public int getMaxZoomLevel() {
      return 22;
   }

   /**
    * Get the zoom level for the map. This may be explicitly set through setZoomLevel()
    * or calculated by prepareCoordinate().
    */
   public int getZoomLevel() {
      return zoomLevel;
   }

   /**
    * Set the zoom level for the map. This will be used for the map if this map service
    * requires discrete zoom level.
    */
   public void setZoomLevel(int zoomLevel) {
      this.zoomLevel = zoomLevel;
   }

   /**
    * Check and adjust latitude/longitude range if necessary.
    */
   public void prepareCoordinate(GeoCoord coord, int width, int height) {
      // no op
   }

   private int zoomLevel = -1;
}
