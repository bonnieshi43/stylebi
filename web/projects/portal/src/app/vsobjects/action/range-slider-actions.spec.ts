/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { TestUtils } from "../../common/test/test-utils";
import { ComposerContextProviderFactory, ViewerContextProviderFactory } from "../context-provider.service";
import { RangeSliderActions } from "./range-slider-actions";

describe("RangeSliderActions", () => {
   const createModel = () => TestUtils.createMockVSRangeSliderModel("VSRangeSlider1");

   //#10281
   it("check status of menu actions and toolbar actions of range slider in composer", () => {
      const expectedMenu = [
         [
            { id: "range-slider properties", visible: true },
            { id: "range-slider viewer-advanced-pane", visible: false },
            { id: "range-slider show-format-pane", visible: true },
            { id: "range-slider edit-range", visible: true },
            { id: "range-slider convert-to-selection-list", visible: false }
         ],
         [
            { id: "vs-object remove", visible: false },
            { id: "range-slider viewer-remove-from-container", visible: false }
         ],
         [
            { id: "vs-object copy", visible: true },
            { id: "vs-object cut", visible: true },
            { id: "vs-object remove", visible: true },
            { id: "vs-object group", visible: true },
            { id: "vs-object ungroup", visible: true }
         ],
         [
            { id: "vs-object bring-forward", visible: true },
            { id: "vs-object bring-to-front", visible: true },
            { id: "vs-object send-backward", visible: true },
            { id: "vs-object send-to-back", visible: true }
         ]
      ];

      const expectedToolbar = [
         [
            //need update later
            { id: "range-slider unselect", visible: true },
         ]
      ];

      const model = createModel();
      const actions = new RangeSliderActions(model, ComposerContextProviderFactory());
      const menuActions = actions.menuActions;
      const toolbarActions = actions.toolbarActions;

      expect(menuActions).toMatchSnapshot();
      expect(toolbarActions).toMatchSnapshot();

      //check composite value
      model.composite = true;
      expect(menuActions[0].actions[3].visible()).toBeFalsy();
   });

   //#17058
   it("check status of menu actions and toolbar actions of range slider in viewer and preview", () => {
      const expectedMenu = [
         [
            { id: "range-slider properties", visible: false },
            { id: "range-slider viewer-advanced-pane", visible: false },
            { id: "range-slider show-format-pane", visible: false },
            { id: "range-slider edit-range", visible: true },
            { id: "range-slider convert-to-selection-list", visible: false }
         ],
         [
            { id: "vs-object remove", visible: false },
            { id: "range-slider viewer-remove-from-container", visible: false }
         ],
         [
            { id: "vs-object copy", visible: false },
            { id: "vs-object cut", visible: false },
            { id: "vs-object remove", visible: false },
            { id: "vs-object group", visible: false },
            { id: "vs-object ungroup", visible: false }
         ],
         [
            { id: "vs-object bring-forward", visible: false },
            { id: "vs-object bring-to-front", visible: false },
            { id: "vs-object send-backward", visible: false },
            { id: "vs-object send-to-back", visible: false }
         ]
      ];

      const expectedToolbar = [
         [
            //need update later
            { id: "range-slider unselect", visible: true },
         ]
      ];

      //check status in viewer
      const model = createModel();
      const actions1 = new RangeSliderActions(model, ViewerContextProviderFactory(false));
      const menuActions1 = actions1.menuActions;
      const toolbarActions1 = actions1.toolbarActions;

      expect(menuActions1).toMatchSnapshot();
      expect(toolbarActions1).toMatchSnapshot();

      //check status in preview
      const model2 = createModel();
      const actions2 = new RangeSliderActions(model2, ViewerContextProviderFactory(true));
      const menuActions2 = actions2.menuActions;
      const toolbarActions2 = actions2.toolbarActions;

      expect(menuActions2).toMatchSnapshot();
      expect(toolbarActions2).toMatchSnapshot();

      //#18021, check status in preview when triggered by adhoc filter
      const expectedToolbar2 = [
         [
            { id: "range-slider unselect", visible: true },
         ]
      ];
      model2.adhocFilter = true;

      expect(toolbarActions2).toMatchSnapshot();

      //#18141, check edit action when in container
      model2.adhocFilter = false;
      model2.container = "aa";
      model2.containerType = "VSSelectionContainer";
      expect(menuActions2[0].actions[3].visible()).toBeTruthy();
      // TODO fix spec
      // expect(menuActions2[0].actions[1].visible()).toBeTruthy();
      // expect(menuActions2[1].actions[1].visible()).toBeTruthy();

      //check status when container not support remove child
      model2.supportRemoveChild = false;
      expect(menuActions2[0].actions[3].visible()).toBeTruthy();
      expect(menuActions2[0].actions[2].visible()).toBeFalsy();
      expect(menuActions2[1].actions[1].visible()).toBeFalsy();
   });

   //bug #18022, #18105, Bug #20859
   it("check status of menu actions and toolbar actions of range slider in composer when in selection container", () => {
      const expectedMenu = [
         [
            { id: "range-slider properties", visible: true },
            { id: "range-slider viewer-advanced-pane", visible: false },
            { id: "range-slider show-format-pane", visible: true },
            { id: "range-slider edit-range", visible: true },
            { id: "range-slider convert-to-selection-list", visible: true }
         ],
         [
            { id: "vs-object remove", visible: true },
            { id: "range-slider viewer-remove-from-container", visible: false }
         ]
      ];

      const expectedToolbar = [
         [
            { id: "range-slider unselect", visible: true },
         ]
      ];

      const model = createModel();
      model.container = "aa";
      model.containerType = "VSSelectionContainer";
      const actions = new RangeSliderActions(model, ComposerContextProviderFactory());
      const menuActions = actions.menuActions;
      const toolbarActions = actions.toolbarActions;

      expect(menuActions).toMatchSnapshot();
      expect(toolbarActions).toMatchSnapshot();

      //bug#18046, viewer remove action
      const actions2 = new RangeSliderActions(model, ViewerContextProviderFactory(false));
      const menuActions2 = actions2.menuActions;
      expect(menuActions2[1].actions[0].visible()).toBeFalsy();
      expect(menuActions2[1].actions[1].visible()).toBeTruthy();

      //check composite value
      model.composite = true;
      expect(menuActions[0].actions[3].visible()).toBeFalsy();
      //expect(menuActions[1].actions[3].visible()).toBeFalsy();
   });

   //Bug #19986 should not display menu action when as data tip component
   it("should not display menu action when as data tip component", () => {
      const dataTipService: any = { isDataTip: jest.fn() };
      dataTipService.isDataTip.mockImplementation(() => true);
      const model = createModel();
      const actions = new RangeSliderActions(model, ViewerContextProviderFactory(false), false, null, dataTipService);
      const menuActions = actions.menuActions;

      expect(menuActions.length).toBe(0);
   });
});