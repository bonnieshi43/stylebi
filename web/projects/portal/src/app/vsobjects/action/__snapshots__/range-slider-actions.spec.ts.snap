// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in composer 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in composer 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in composer when in selection container 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in composer when in selection container 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in viewer and preview 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in viewer and preview 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in viewer and preview 3`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in viewer and preview 4`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`RangeSliderActions check status of menu actions and toolbar actions of range slider in viewer and preview 5`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;
